[Muhammad Assyarul Ariffin] iP

Assyarul opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

Assyarul committed:
initial level-1

Assyarul committed:
Level 2 initial

Assyarul committed:
Level-3 w/ Command and Decorator Patterns implemented

Assyarul committed:
Level 4 w/ Factory Pattern for Task subtypes

Assyarul committed:
Level-5. Error-handling.

Assyarul committed:
Level-6. Added delete functionality.

Assyarul committed:
Level-7 Save

Assyarul committed:
Level-8 Date and Time parse

Assyarul committed:
Merge branch 'branch-Level-7' into master

Assyarul committed:
Fix merge conflict with branch-Level-8

Assyarul committed:
Added more OOP

Assyarul committed:
Add JUnit test for EventFactory class

Assyarul committed:
Added JavaDocs to some classes

Assyarul committed:
Level-9 added Find feature

Assyarul committed:
Fix bug where file has not been initially read when program started

Assyarul committed:
Update modifiers on most methods and classes.

Assyarul committed:
Merge branch 'branch-A-JavaDoc' into master

Assyarul committed:
Fix merge conflict between 'branch-A-CodingStandard' and master

Assyarul committed:
Fix merge conflict with branch-Level-9 and master

Assyarul committed:
refactor: implemented packages.

fix:fixed minor bug in Storage.

Assyarul committed:
docs: added more javadocs to duke.command package.

This commit also contains changes that was supposed
to be brought forth from previous commit.

Assyarul committed:
refactor: restructure test cases to be in separate test folder

Assyarul committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2021S2/ip into master

Assyarul committed:
Implement gradle usage for automation,

Current state does not use gradle for automation.

Since there's no automation in place, it is harder to
check the consistency of the code.

Let's implement gradle into the project. Additionally,
code will be changed to avoid checkstyle errors when
building using gradle.

Stratostorm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,69 @@
+            }
+            reader.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("No existing file found.");
```
Maybe consider moving the printing of error messages to be handled by the UI?

Stratostorm commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    private void initialiseCommandMap() {
+
```
Extra blank line here

Stratostorm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * Attempts to read the file named "duke.txt". Will print out
```
Maybe move "Will print out" to the next line

Stratostorm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Attempts to write to the file named "duke.txt". Will throw
+     * IOException if file could not be written.
+     * @throws IOException
```
Add an empty line here?

Stratostorm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+ * Class handling List of Tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> list;
```
Maybe use 'tasks' instead of list.

Stratostorm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+
+    /**
+     * Gets the ArrayList of Tasks.
+     * @return ArrayList of Tasks
```
Could add empty line between description and parameters section, here and in other places

Stratostorm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+
+    /**
+     * Deletes the task at index.
+     * @param count the index of task to be deleted.
```
Would it be better to change 'count' to 'index'?

Stratostorm commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,44 @@
+ * Handles the interaction the user and the program.
+ */
+class Ui {
+    private Scanner scanner;
```
Could add an empty line here between the variables and constructor.

Stratostorm commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,44 @@
+        scanner = new Scanner(System.in);
+    }
+
+    protected String getLine() throws NoSuchElementException {
```
I think getNextLine or even just nextLine would be a better name, getLine implies getting a specific line from a line number

Stratostorm commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    /**
+     * Command that prints the list existing in duke.TaskList.
```
This method doesn't print the message, just returns it correct?

Stratostorm commented on src/main/java/duke/command/CommandBorder.java:
```diff
@@ -0,0 +1,28 @@
+    private ICommand decoratedCommand;
+
+    /**
+     * Generate Tasks depending on the type needed.
```
Sorry, I don't get what this does...

Stratostorm commented on src/main/java/duke/command/DefaultCommand.java:
```diff
@@ -0,0 +1,17 @@
+ */
+public class DefaultCommand implements ICommand {
+    /**
+     * When executed, will print a default statement. Input given will not change
```
Maybe move 'Input given will not change' to the same line as 'how function gets executed.' Noticed it in some other places as well

Assyarul committed:
add GUI to Duke. Fix major bug in TaskList."

Assyarul committed:
Merge branch 'branch-Level-10' into master

Assyarul committed:
Change to Date Parser Format to dd MMM yyyy

Currently, state of parser given uses MMM dd yyyy which
is incompatible with Singapore default style of displaying dates.

Let's change to the correct format.

Assyarul committed:
Add Assertions and more JavaDocs

No Assertions has been implemented in current version
and some JavaDocs were incomplete.

Let's add some Assertions in DialogBox and Ui, ensuring
FXML files are loaded correctly.

Assyarul committed:
Add Assertions and more JavaDocs

No Assertions has been implemented in current version
and some JavaDocs were incomplete.

Let's add some Assertions in DialogBox and Ui, ensuring
FXML files are loaded correctly.

Assyarul committed:
Improve Code Quality of Duke.

Certain methods can be improved by moving some of code to
Parser due to the nature of the code (which parses mostly).

Let's move some of these codes to become static methods in
the class Parser.

Assyarul committed:
Amend previous commit

Assyarul committed:
Removes merge conflict artifact.

Assyarul committed:
Add use of Lambda and Streams for finding keywords.

Currently using for loop to find all the task with the
relevant keywords.

Let's change the implemention to use a one-liner through
the use of Lambdas and Streams.

Assyarul committed:
Merge pull request #1 from Assyarul/branch-A-Lambda

Add use of Lambda and Streams for finding keywords and add functionality to ScrollPane UI.

Assyarul committed:
Merge pull request #2 from Assyarul/branch-A-CodeQuality

Improve Code Quality of Duke

Assyarul committed:
Merge branch 'master' into branch-A-Assertions

Assyarul committed:
Merge pull request #3 from Assyarul/branch-A-Assertions

Add assertions to do sanity checks, add JavaDocs and add an unhandled exception in Storage.java

Assyarul committed:
Add C-Priority extension.

Currently there is no priority system for the tasks.

Let's add a rudimentary priority system for the tasks.

Assyarul committed:
Merge pull request #4 from Assyarul/branch-C-Priority

Add C-Priority extension.

Assyarul committed:
Add JUnit tests, fix CheckStyle Errors, minor bugs.

Todo tasks have been showing up twice when list is executed.
Additionally, there were still some CheckStyle Errors that exist.
There are also very few JUnit test classes in the code.

Let's add more JUnit test classes, fix the remaining CheckStyle errors
and fix the minor bug.

Assyarul committed:
Add some borders to the DialogBox Labels. Add color to error messages

Assyarul committed:
Merge branch 'branch-A-BetterGui'

Assyarul committed:
Add User Guide & minor bugfixes

Assyarul committed:
Set theme jekyll-theme-cayman

Assyarul committed:
Update User Guide

Assyarul committed:
Merge branch 'master' of https://github.com/Assyarul/ip

Assyarul committed:
Add Table of Content to User Guide

Assyarul committed:
Minor change to User Guide

Assyarul committed:
Fix User Guide Table of Content

Assyarul committed:
Change User Guide Table of Contents
